### ISSUE TITLE
Application crashing while connecting to Bluetooth headphones | Windows 11

### ISSUE THREAD

**Issue by saahil1292** [fxsound2/fxsound-app#166]
**Describe the bug**
Whenever I connect my Bluetooth headphones, sometimes it doesn't appear in the list. And whenever it does, and I select it and then adjust the volume from sound in the taskbar, the application crashes. This did not happen in the previous version and with Windows 10. Currently using Windows 11.

This behaviour usually happens after I restart my system.

**To Reproduce**
1. Go to FX Sound application
2. Click on device dropdown and select Bluetooth headphones
3. Go to sound settings to adjust volume
4. Application crashes with no sound coming

**Expected behavior**
No application crashing while adjusting sound, system auto detect Bluetooth headphones properly

**Screenshots**
NA

**System information**
 - Windows version - Windows 11 10.0.22631
 - Audio device - Bluetooth Headphones (any)
 - FxSound version - 1.1.27

**Additional context**
NA

**Comment by doolhoofd** [comment]
See https://github.com/fxsound2/fxsound-app/issues/21
Please consult 
https://forum.fxsound.com/t/troubleshooting/3105
first, and post back with your results.

> 9) If the problematic device connects via Bluetooth, try factory-resetting it or clearing the device’s Bluetooth memory, and completely removing, then re-adding it to your computer’s Bluetooth list;

**Comment by saahil1292** [comment]
I have removed the device from the device manager and after restarting again added the device. The problem still persists.

**Comment by doolhoofd** [comment]
Our apologies for this unintended bug.

The numerous possible combinations of components involved in any digital audio setup, like varying hardware and software elements, different input and output devices, elaborate settings menus, drivers, operating system versions, and any unknown and unexpected conflicts or incompatibilities that might arise between them, inevitably creates a problem of inherent complexity. As a result, tracing issues down to their root causes and finding a permanent solution, is often a challenging process. 
As I mentioned, we are aware of the fact that using FxSound together with Bluetooth creates varying issues for many users, and we are committed to improving FxSound's Bluetooth functionalities as one of our top priorities. 
However, offering you a quick and permanent fix for your particular problem, either through extended troubleshooting or through coding efforts, does not seem to be possible at the current time.

Therefore, these would appear to be your options:

- Be sure to set the Bluetooth device in use as the Default Communication Device;

- [Manually remove the FxSound Audio Enhancer driver from Windows Device Manager](https://forum.fxsound.com/t/echo-due-to-fx-and-the-original-playback-device-playing-at-the-same-time/2112/2), and download and install a [previous version](https://www.videohelp.com/software/FxSound-/old-versions);

- Power Off FxSound, or disable the FxSound Speakers device in Windows Sound Settings, while using a Bluetooth device, and try boosting its audio using the built-in equalizer and effects settings of your media player (VLC, WinAmp, Windows Media Player, MusicBee, Neutron), or a browser extension like the options linked at the end of [the main troubleshooting list](https://forum.fxsound.com/t/troubleshooting/3105) instead. You can keep the FxSound Speakers device disabled as long as necessary without any harmful effects, except for the "Oops!" pop-up, which you can safely ignore in this case;

- Wait for the official release of v1.1.27.0, and update upon release;

- Switch to a different audio enhancement option.

Once again, our apologies. Thank you for your understanding.

**Comment by bvijay74** [comment]
Hi @saahil1292 
Sorry about the crash that you are facing.
Is FxSound crashing when you connect a specific Bluetooth device or any Bluetooth device? Can you please share the device details which causes the crash and the settings file located at %APPDATA%\FxSound\FxSound.settings

**Comment by saahil1292** [comment]
Hi, thanks for the response. I really love FXSound and want to keep it using for my default sound enhancer. @doolhoofd I was looking for the old version but couldn't find in the Github repo. @bvijay74 please find the settings file as requested.

Note: - I disabled the application on Windows Startup, then connected the Bluetooth device, and then started the application and so far, it successfully runs without any crash. Perhaps one thought, the issue might lie
 with the sync of windows volume bar and FXSound sound bar graphic?

I'll try to use the older version which doesn't have the graphic and see if the hypothesis is correct or not.

[FxSound.txt](https://github.com/user-attachments/files/16832757/FxSound.txt)

**Comment by raddyst** [comment]
> I was looking for the old version but couldn't find in the Github repo.

You can download any (opensource) old version from https://github.com/fxsound2/fxsound-app/releases, the installer can be found in fxsound-app-*.zip/fxsound-app-*/release folder

**Comment by saahil1292** [comment]
UPDATE: - Okay, same thing happened with 1.1.26 version as well. One thing I noticed is if the FXSound application starts with the Bluetooth device as output, no issues occur even if I enable the application in Windows Startup. However, If I switch to it afterwards, it crashes. 

I generally have a speaker attached with AUX as my default output device. I switch to my Bluetooth headphones when playing games or working on editing stuff.

**Comment by bvijay74** [comment]
Hi Saahil,
I tried to replicate the crash scenario. I used your FxSound settings file and changed the output device id and name to the Bluetooth headphone that I use. I tried starting FxSound with the Bluetooth headphone as the output and also connecting and switching to Bluetooth heaphone after starting FxSound. In both scenarios I did not encounter any crash.

Can you please confirm if you have any other audio enhancement application running? 
Also please check in Control Panel -> Sound, if any enhancement option is enabled in the device properties and if exclusive mode is enabled for the device in the Advanced tab.
What is the default audio format set for your device?

**Comment by saahil1292** [comment]
Hi @bvijay74 ,

Please find below the sound settings and default audio format.

I do not have any other audio enhancement software installed apart from FXSound.

![image](https://github.com/user-attachments/assets/942c45ab-b56e-48b7-96a9-2cf81d7f2594)

**Comment by bvijay74** [comment]
Hi Saahil,
Thanks for sharing this information.
Sorry for not making it clear. Can you please check the properties of the Bluetooth device. And see if you find this kind of options are present and enabled,
![image](https://github.com/user-attachments/assets/82140c00-9867-4f64-ab79-57e432ea08f4)
![image](https://github.com/user-attachments/assets/3acf7476-a9e6-4b30-808d-e7e31025e359)

You can also disable exclusive mode in FxSound Speakers.

**Comment by saahil1292** [comment]
@bvijay74 Nope, just simple general tab is present. Nothing else.

![image](https://github.com/user-attachments/assets/d2cace69-0094-4e50-a3ec-2bcb66543ba3)

PS: This is happening with all the versions. I tried using the 1.17 version as well, it is still crashing.

**Comment by bvijay74** [comment]
Hi @saahil1292 
Can you see the audio formats for the device through System  > Sound?
![image](https://github.com/user-attachments/assets/a73e5d3a-d637-4425-b8cd-d1a1064d94a9)


FxSound can fail if it cannot get the output device properties.

**Comment by saahil1292** [comment]
@bvijay74 Yes, I do get them.

![image](https://github.com/user-attachments/assets/1cdfc1e9-edbe-471e-a948-ae466acb1f51)

**Comment by bvijay74** [comment]
Hi @saahil1292 
If your Bluetooth headphone supports 2 channel stereo mode, can you set it to stereo.
FxSound does not support single channel devices.
But FxSound should have not listed the 1 channel device instead of crashing. I will investigate that.

**Comment by saahil1292** [comment]
@bvijay74 Yes, I believe it is switching back to 1 channel and then it is not able to connect.

Right now, my headphones are connected to FXSound and working fine after I had to switch Bluetooth on and off couple of times and it is showing 2 channels.

![image](https://github.com/user-attachments/assets/4836dcce-f38e-4d99-bc8e-d8fa78d188e8)
